libs/utils/utils.js

```diff
@@ -562,7 +562,9 @@ export function decorateAutoBlock(a) {
       return false;
     }
 
-    if (key === 'fragment') {
+    const nameSplit = a.href.split('/').pop().split('.');
+    const mp4Match = 'media_.*.mp4';
+    if (key === 'fragment' && (nameSplit.length <= 1 || a.textContent.match(mp4Match))) {
```

Comment:
Regular expressions can be costly, so I'd avoid calling `match` two times, once here and once on L584. Could we cache the result once and use it later as well?

Reply:
@overmyheadandbody That's a good point.  Tweaked to save results and only run the match once.

Reply:
Thanks for looking into this!